src/views/components/map/Map.web.js

```diff
@@ -1,16 +1,510 @@
+                  icon: this.props.mapRouteIcon,
+                  map,
+                });
+
+                this.markers.push({
+                  newMarker,
+                });
+
+                newMarker.addListener( 'click', () => {
+                  infowindow.open( map, newMarker );
+                  infowindow.setContent( route.origin );
+                });
+
+                const originCoords = new google.maps.LatLng(
+                  routeOriginCoords.lat, routeOriginCoords.lng 
+                );
+
+                this.locations.push( originCoords );
+
+                this.checkAddressFormat( geocoder, route.dest, ( routeDestCoords ) => {
+                  const newMarker = new google.maps.Marker({
+                    position: {
+                      lat: parseFloat( routeDestCoords.lat ),
+                      lng: parseFloat( routeDestCoords.lng ),
+                    },
+                    zIndex: 100,
+                    icon: this.props.mapRouteIcon,
+                    map,
+                  });
+
+                  this.markers.push({
+                    newMarker,
+                  });
+
+                  newMarker.addListener( 'click', () => {
+                    infowindow.open( map, newMarker );
+                    infowindow.setContent( route.dest );
+                  });
+
+                  const destCoords = new google.maps.LatLng(
+                    routeDestCoords.lat, routeDestCoords.lng 
+                  );
+
+                  this.locations.push( destCoords );
+                  this.calcRoute( originCoords, destCoords, () => {
+                    if ( counterRoutes === routes.length - 1 ) {
+                      adjustMapBounds();
+                    }
+
+                    counterRoutes += 1;
+                  });
+                });
+              });
+            });
+          });
+        }
+        catch ( error ) {
+          this.isError = true;
+          this.setAlert( 'Map could not be loaded. Please contact support.' );
+        }
+      }
+    }
+
+    getLatLng = () => {
+      const { center, lat, lng } = this.props;
+
+      let newLat = center.lat;
+      let newLng = center.lng;
+
+      if ( lat && lng ) {
+        if ( typeof lat !== 'number' ) {
+          const checkLat = parseFloat( lat );
+
+          if ( typeof lng !== 'number' ) {
+            const checkLng = parseFloat( lng );
+
+            if ( Number.isInteger( checkLat ) && Number.isInteger( checkLng )) {
+              newLat = parseFloat( checkLat );
+              newLng = parseFloat( checkLng );
+            }
+          }
+        }
+      }
+
+      return { lat: newLat, lng: newLng };
+    }
+
+    componentDidMount() {
+      this.setup();
+      this.drawPolygons();
+      this.drawPolylines();
+    }
+
+    checkAddressFormat = ( geocoder, value, callback ) => {
+      let address = value;
+
+      if ( typeof address === 'string' || 
+      ( typeof address === 'object' && 
```

Comment:
Can we please use our internal `isObject` here, and use the options called `withProperties` and pass through `['suburb', 'state]`?